### PR TITLE
fix: Unwrap nested embedded structs

### PR DESCRIPTION
**Pull request by candiduslynx**
fixes https://github.com/cloudquery/cloudquery-issues/issues/1695

### CODE REVIEW

File: transformers/struct.go [hunk 1]
```diff
@@ -228,9 +225,10 @@ func (t *structTransformer) ignoreField(field reflect.StructField) bool {
 	switch {
 	case len(field.Name) == 0,
 		slices.Contains(t.skipFields, field.Name),
-		!field.IsExported(),
 		isTypeIgnored(field.Type):
 		return true
+	case !field.IsExported():
```
Comment (candiduslynx) [inline transformers/struct.go:230]: this is required as the embedded structs might contain other embedded structs and so on

File: scalar/date32_test.go [hunk 2]
```diff
@@ -5,29 +5,27 @@ import (
...
 func TestDate32Set(t *testing.T) {
 	successfulTests := []struct {
 		source any
 		result Date32
 	}{
-		{source: time.Date(1900, 1, 1, 0, 0, 0, 0, time.Local), result: Date32{Value: arrow.Date32FromTime(time.Date(1900, 1, 1, 0, 0, 0, 0, time.Local)), Valid: true}},
-		{source: time.Date(1999, 12, 31, 12, 59, 59, 0, time.Local), result: Date32{Value: arrow.Date32FromTime(time.Date(1999, 12, 31, 0, 0, 0, 0, time.Local)), Valid: true}},
+		{source: time.Date(1900, 1, 1, 0, 0, 0, 0, time.UTC),
+			result: Date32{Value: arrow.Date32FromTime(time.Date(1900, 1, 1, 0, 0, 0, 0, time.UTC)), Valid: true}},
+		{source: time.Date(1999, 12, 31, 12, 59, 59, 0, time.UTC),
```
Comment (candiduslynx) [inline scalar/date32_test.go:18]: this was failing on my machine/locality, so moved to utc

**Comment by disq** [comment]
This could be opt-in in the first release. Can you check if there are high number of plugins affected by this change, other than Slack?

**Comment by candiduslynx** [comment]
> This could be opt-in in the first release. Can you check if there are high number of plugins affected by this change, other than Slack?

@disq 
https://github.com/cloudquery/cloudquery-private/pull/2489

**Comment by erezrokah** [comment]
Can we close this PR until we can find the time to discuss the default behavior we want via the issue?

**Pull request closed without merging**